dask_glm/estimators.py

```diff
@@ -0,0 +1,137 @@
+"""
+Models following scikit-learn's estimator API.
+"""
+from sklearn.base import BaseEstimator
+
+from . import algorithms
+from . import families
+from .utils import sigmoid, dot, add_intercept
+
+
+class _GLM(BaseEstimator):
+    @property
+    def family(self):
+        """
+        Family
+        """
+
+    def __init__(self, fit_intercept=True, solver='admm', regularizer='L1',
+                 max_iter=100, tol=1e-4, lamduh=0.1, rho=1,
+                 over_relax=1, abstol=1e-4, reltol=1e-2):
+        self.fit_intercept = True
+        self.solver = solver
+        self.regularizer = regularizer
+        self.max_iter = max_iter
+        self.tol = tol
+        self.lamduh = lamduh
+        self.rho = rho
+        self.over_relax = over_relax
+        self.abstol = abstol
+        self.reltol = reltol
+
+        self.coef_ = None
+        self.intercept_ = None
+        self._coef = None  # coef, maybe with intercept
+
+        fit_kwargs = {'max_iter', 'tol', 'family'}
+
+        if solver == 'admm':
+            fit_kwargs.discard('tol')
+            fit_kwargs.update({
+                'regularizer', 'lamduh', 'rho', 'over_relax', 'abstol',
+                'reltol'
+            })
+        elif solver == 'proximal_grad':
+            fit_kwargs.update({'regularizer', 'lamduh'})
+
+        self._fit_kwargs = {k: getattr(self, k) for k in fit_kwargs}
+
+    def fit(self, X, y=None):
```

Comment:
The only other call-out I'll make here is that this initialization setup will make it difficult for a user to create an unregularized model (they will need to specify a hand-picked solver).  I'm not worried about it for this PR, but I think we should be aware of it and make it easier (without having the user choose the solver) in the future.